src/values/AbstractValue.js

```diff
@@ -193,6 +193,12 @@ export default class AbstractValue extends Value {
       }
       return true;
     };
+    // when we have a ToObject conversion, we don't want to
+    // the equivalence to equal one another as they may have
+    // been created at different points
+    if (this.kind === "explicit conversion to object") {
```

Comment:
Is this the best place to fix it? Or is there some more appropriate fix higher up the call stack?

Technically I’d say they’re still “equal”. Equality is just not sufficient in this case to allow replacing them. As Seb noted maybe they should’ve been temporal? Or maybe something above can say “let’s rely on equality unless the timing aspect matters”.